core/src/main/java/org/elasticsearch/cluster/health/ClusterShardHealth.java

```diff
@@ -59,6 +62,9 @@ public ClusterShardHealth(final int shardId, final IndexShardRoutingTable shardR
             } else if (shardRouting.unassigned()) {
                 computeUnassignedShards++;
             }
+            if (shardRouting.primary()) {
+                primaryRouting = shardRouting;
+            }
```

Comment:
Simpler just to write `ShardRouting primaryRouting = shardRoutingTable.primaryShard();` which is guaranteed non-null.

Let's also remove `computePrimaryActive` and directly write `primaryRouting.active()` instead.